tests/components/homekit/test_type_thermostats.py

```diff
@@ -676,6 +679,124 @@ def cls():
     assert acc.char_target_heat_cool.value == 1
 
 
+async def test_thermostat_hvac_modes_with_auto_heat_cool(hass, hk_driver, cls):
+    """Test we get heat cool over auto."""
+    entity_id = "climate.test"
+
+    hass.states.async_set(
+        entity_id,
+        HVAC_MODE_HEAT_COOL,
+        {
+            ATTR_HVAC_MODES: [
+                HVAC_MODE_HEAT_COOL,
+                HVAC_MODE_AUTO,
+                HVAC_MODE_HEAT,
+                HVAC_MODE_OFF,
+            ]
+        },
+    )
+    call_set_hvac_mode = async_mock_service(hass, DOMAIN_CLIMATE, "set_hvac_mode")
+    await hass.async_block_till_done()
+
+    acc = cls.thermostat(hass, hk_driver, "Climate", entity_id, 2, None)
+    await hass.async_add_job(acc.run)
```

Comment:
`hass.async_add_job` is legacy. Use `hass.async_add_executor_job` if this should run in the executor.

Reply:
Its actually calling it again under the hood as well
https://github.com/home-assistant/core/pull/33727